libs/contrib/Decidable/Decidable/Extra.idr

```diff
@@ -47,3 +51,49 @@ doubleNegationExists {ts} {r} @{dec} nnxs =
       witnessing   : uncurry              r  witness
       witnessing   = doubleNegationElimination @{dec} witness witnessingnn
   in introduceWitness witness witnessing
+
+
+
+
+decideTransform :
+  {n : Nat}
+  -> {ts : Vect n Type}
+  -> {r : Rel ts}
+  -> {t : Type -> Type}
+  -> (tDec : {a : Type} -> Dec a -> Dec (t a))
+  -> (posDec : Decidable n ts r)
```

Comment:
If this is not an `auto` argument then is it better to use the
interface or rather to use `IsDecidable` instead? I think building
dictionaries may be a bit more annoying than just passing the
function directly?

e.g. in `Data.List` we have:

```idris
lookupBy : (a -> a -> Bool) -> a -> List (a, b) -> Maybe b
```
and not

```idris
lookupBy : (eq : Eq a) -> a -> List (a, b) -> Maybe b
```